src/web/app/components/extension-confirm-modal/extension-confirm-modal.component.spec.ts

```diff
@@ -141,16 +142,168 @@ describe('ExtensionConfirmModalComponent', () => {
   it('use ngOnInit to initialise', () => {
     component.selectedStudents = [studentModel3];
     component.selectedInstructors = [instructorModel3];
+    const setStudentTableDataSpy = jest.spyOn(component, 'setStudentTableData');
+    const setInstructorTableDataSpy = jest.spyOn(component, 'setInstructorTableData');
     component.ngOnInit();
     component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
     fixture.detectChanges();
     expect(fixture).toMatchSnapshot();
+    expect(setStudentTableDataSpy).toHaveBeenCalled();
+    expect(setInstructorTableDataSpy).toHaveBeenCalled();
+  });
+
+  it('ngOnInit to initialise with empty arrays', () => {
+    component.selectedStudents = [];
+    component.selectedInstructors = [];
+    const setStudentTableDataSpy = jest.spyOn(component, 'setStudentTableData');
+    const setInstructorTableDataSpy = jest.spyOn(component, 'setInstructorTableData');
+    component.ngOnInit();
+    component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
```

Comment:
let's not use the snapshots here, while it would be nice to have snapshots every test, we should keep it only for non-edge cases as they are expensive to run